backend/app/auth.js

```diff
@@ -185,12 +189,55 @@ passport.use(new JwtStrategy(jwtOptions,
                 'WHERE "Users"."email" = \'' + decodedJWTPayload.email  +'\'' +
                 ') '
             );
-
-            if (!_.first(user)) {
-                throw new HttpError(404, 'No user found');
+            
+            //TODO: Rather than checking the public schema, decide on how scopes from the
+            //TODO: auth service will work with indaba and modify this
+            if (!_.first(user)) { // user doesn't exist in the given realm
+                if (req.params.realm !== config.pgConnect.adminSchema) { //look in the Public realm if the user is a superAdmin
+                    adminUser = yield admThunkQuery(
+                        User
+                            .select(
+                                User.star(),
+                                Role.name.as('role')
+                            )
+                            .from(
+                                User
+                                    .leftJoin(Role)
+                                    .on(User.roleID.equals(Role.id))
```

Comment:
Right now if a user exist on the auth service and doesn't exist on greyscale this will throw a 
`Cannot read property 'id' of undefined` error. Eventually when we modify to work with scopes, we'll get rid of this.